pycqed/instrument_drivers/meta_instrument/qubit_objects/mock_CCL_Transmon.py

```diff
@@ -60,164 +62,284 @@ def add_mock_params(self):
+        self.add_parameter('mock_cfg_dc_flux_ch',
+                           label='most closely coupled fluxline',
+                           unit='', initial_value='FBL_1',
                            parameter_class=ManualParameter)
 
-    def find_resonators_VNA(self, start_freq=7e9, stop_freq=8e9, power=-40,
-                            bandwidth=200, timeout=200, npts=50001, name='VNA'):
-        VNA = self.instr_VNA.get_instr()
-        t_start_meas = ma.a_tools.current_timestamp()
+        self.add_parameter('noise', label='nominal noise level', unit='V',
+                           initial_value=0.16e-3, parameter_class=ManualParameter)
 
-        VNA.start_frequency(start_freq)
-        VNA.stop_frequency(stop_freq)
-        VNA.power(power)
-        VNA.bandwidth(bandwidth)
-        VNA.npts(npts)
-        # VNA.timeout(timeout)
+    def find_resonators(self, start_freq=7e9, stop_freq=8e9, power=-40,
+                        bandwidth=200, timeout=200, npts=2001, with_VNA=None,
+                        verbose=True):
+
+        if with_VNA is None:
+            try:
+                if self.instr_VNA.get_instr() == '':
+                    with_VNA = False
+                else:
+                    with_VNA = True
+            except:
+                with_VNA = False
 
-        name = 'Initial_VNA'
+        if with_VNA:
+            VNA = self.instr_VNA.get_instr()
 
-        self.measure_with_VNA(VNA, name=name)
+            VNA.start_frequency(start_freq)
+            VNA.stop_frequency(stop_freq)
+            VNA.power(power)
+            VNA.bandwidth(bandwidth)
+            VNA.npts(npts)
+            # VNA.timeout(timeout)
+            name = 'Initial_VNA'
 
-        result = sa2.Initial_VNA_Analysis(label=name)
+            self.measure_with_VNA(VNA, name=name)
+
+            result = sa2.Initial_Resonator_Scan_Analysis(label=name)
+        else:
+            self.ro_pulse_amp_CW(0.01)
+            freqs = np.linspace(start_freq, stop_freq, npts)
+            self.measure_heterodyne_spectroscopy(freqs=freqs, analyze=False)
+            result = sa2.Initial_Resonator_Scan_Analysis()
 
         peak_freqs = []
         for peak in result.peaks:
             if peak not in peak_freqs:
                 peak_freqs.append(peak)
 
-        self.resonator_freqs(result.peaks)
-
         self.res_dict = {}
         for i, freq in enumerate(result.peaks):
-            self.res_dict[str(i)] = ['unknown', freq, {}, None]
+            self.res_dict[str(i)] = [freq, 'unknown', {}, None, 0]
+
+        if verbose:
+            for resonator, items in self.res_dict.items():
+                print('{}:\t{:.3f} GHz'.format(resonator, items[0]/1e9))
 
-        print(self.resonator_freqs())
         return True
 
-    def find_resonator_frequency_VNA(self, start_freq=7e9, stop_freq=8e9,
-                                     use_min=False, MC=None, update=True):
+    def calibrate_spec_pow(self, freqs=None, start_power=None, verbose=True):
```

Comment:
The calibrate method should not be defined in the mock qubit, instead the measure should be defined in the mock and the routine making use of it defined in the qubit_object abstract base class.